File: leetcode/026-remove-duplicates-sorted-array.c
int removeDuplicates(int *nums, int numsSize) {
  int prior = nums[0] - 1;
  int i = 0; // trailing pointer
  int j = 0; // leading pointer
  while (j < numsSize) {
    if (nums[j] != prior)
      nums[i++] = nums[j];
    prior = nums[j++];
  }
  return i;
}

File: leetcode/026-remove-duplicates-sorted-array.py
class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        prior = None
        i = 0
        while True:
            try:
                if nums[i] == prior:
                    del nums[i]
                else:
                    prior = nums[i]
                    i += 1
            except IndexError:
                return len(nums)